src/python/pants/backend/python/target_types.py

```diff
@@ -485,6 +485,26 @@ class PythonRequirementLibrary(Target):
 # -----------------------------------------------------------------------------------------------
 
 
+def parse_requirements_file(content: str, *, rel_path: str) -> Iterator[Requirement]:
+    """Parse all `Requirement` objects from a requirements.txt-style file.
+
+    This will safely ignore any options starting with `--` and will ignore comments. Any pip-style
+    VCS requirements will fail, with a helpful error message describing how to use PEP 440.
```

Comment:
It probably makes sense to use Pex's requirements file parser in the future - which handles pip proprietary formats, of which VCS is just one, as well as env var interpolation, line continuations, ... we reject otherwise valid, though perhaps uncommon, requirements files un-neccessarily as it stands.

Reply:
How would we envision that happening? That portion being exposed as a separate library?

Reply:
Let's move that discussion to https://github.com/pantsbuild/pants/issues/11409.